frontend/javascripts/oxalis/model_initialization.js

```diff
@@ -109,12 +109,29 @@ export async function initialize(
     datasetId = { name, owningOrganization };
   }
 
-  const [dataset, initialUserSettings, initialDatasetSettings, tracing] = await fetchParallel(
+  const [dataset, initialUserSettings, tracing] = await fetchParallel(
     annotation,
     datasetId,
     versions,
   );
 
+  let displayedLayers = [];
+  if (dataset.dataSource.dataLayers != null) {
+    displayedLayers = dataset.dataSource.dataLayers.map(layer => ({
+      name: layer.name,
+      isSegmentationLayer: layer.category !== "color",
+    }));
+  }
+
+  if (tracing != null && tracing.volume != null) {
+    if (tracing.volume.fallbackLayer == null) {
+      displayedLayers = displayedLayers.filter(layer => !layer.isSegmentationLayer);
+    }
+    displayedLayers.push({ name: tracing.volume.id, isSegmentationLayer: true });
+  }
```

Comment:
I'm afraid I don't really understand why the front-end needs to send this layer list to the back-end for the dataset settings. Is it only so that the volume layer specific settings can be stored per tracing? Wouldn't it be sufficient to only send the volume tracing id in that case (since the back-end should already know all the layers)?

If this needs to stay, I'd suggest to try to deduplicate this "replace segmentation layer with fallback layer"-logic somehow, since it also appears in `setupLayerForVolumeTracing`.

Reply:
The basic concept is that the frontend can tell the backend, which layer settings it needs, to prevent the old overwriting behavior. However, as the backend only updates the layers sent back from the frontend, it is not necessary to send the complete list, as you mentioned. I included the filtering to not access irrelevant settings in the backend + db, but I think the overhead is not large, so I will adapt it according to your suggestion.